src/compiler/scala/tools/nsc/backend/jvm/opt/BytecodeUtils.scala

```diff
@@ -123,6 +123,8 @@ object BytecodeUtils {
 
   def isReference(t: Type) = t.getSort == Type.OBJECT || t.getSort == Type.ARRAY
 
+  def isScala(classNode: ClassNode) = classNode.attrs != null && classNode.attrs.asScala.exists(a => a.`type` == BTypes.ScalaAttributeName || a.`type` == BTypes.ScalaSigAttributeName)
```

Comment:
The switch to annotation was made in ef1577a9c5a9f2905e9b89b1508f97127d941bd7. I don't know whether leaving the attribute was intentional or an oversight.